crates/rolldown/src/ast_scanner/side_effect_detector/mod.rs

```diff
@@ -248,7 +248,9 @@ impl<'a> SideEffectDetector<'a> {
       Declaration::ClassDeclaration(cls_decl) => self.detect_side_effect_of_class(cls_decl),
       // Currently, using a fallback value to make the bundle correct,
       // finishing the implementation after we carefully read the spec
-      Declaration::UsingDeclaration(_) => true,
+      Declaration::UsingDeclaration(decl) => {
+        decl.is_await || self.detect_side_effect_of_using_declarators(&decl.declarations)
```

Comment:
According to https://github.com/rolldown/rolldown/blob/07c00a0be85ae53fb0b0a3a61e74afe549705746/crates/rolldown/tests/esbuild/dce/dce_of_using_declarations/entry.js#L1-L1